agent/nmap_agent.py

```diff
@@ -253,15 +253,29 @@ def _emit_network_scan_finding(
             technical_detail = (
                 f"{scan_result_technical_detail}\n```xml\n{normal_results}\n```"
             )
-            host = scan_results.get("nmaprun", {}).get("host", {})
-            domains = host.get("hostnames", {})
-            ports = host.get("ports", {}).get("port", "")
-            address = host.get("address", {})
-            if domains:
-                domains = domains.get("hostname", {})
-                if isinstance(domains, List):
-                    for domain_dict in domains:
-                        domain = domain_dict.get("@name", "")
+            hosts = scan_results.get("nmaprun", {}).get("host", {})
+            if isinstance(hosts, dict) is True:
+                hosts = [hosts]
+            for host in hosts:
+                domains = host.get("hostnames", {})
+                ports = host.get("ports", {}).get("port", "")
+                address = host.get("address", {})
+                if domains is not None and len(domains.values()) > 0:
```

Comment:
Reverse the check to decrease the indentations

Reply:
> Reverse the check to decrease the indentations

Yes, & the whole thing can go into a function..

Reply:
I can't reverse this condition, because it is chained with other `elif`s, inside a loop, can't use `continue`